portal2-internal/sdk/misc/color_t.hpp

```diff
@@ -85,38 +88,44 @@ struct color_t {
     }
 
     inline color_t& replace_a(channel_flt_t a) {
-        this->a = channel_t(a * channel_limit_t::max());
+        this->a = static_cast<channel_t>(a * channel_limit_t::max());
         return *this;
     }
 
 public:
     // @credits: https://github.com/znsoftm/xtd/blob/master/src/xtd.drawing/src/xtd/drawing/color.cpp#L186
     [[nodiscard]] static color_t from_hsb(flt_range_t hue, flt_range_t saturation, flt_range_t brightness) {
-        if (!saturation)
-            return color_t(brightness, brightness, brightness, static_cast<channel_flt_t>(1.f));
+        if (!saturation) // @todo: kehrazy: float to bool conversion? isn't this always false in modern c++?
```

Comment:
its not
https://godbolt.org/z/Ejq5WfMed

Reply:
so,

```cpp
if (!safe_cast<int>(saturation)) {
```
?

Reply:
nah, just leave it as it is and remove comment